server/server.go

```diff
@@ -0,0 +1,168 @@
+package server
+
+import (
+	"fmt"
+	"net"
+	"context"
+	"strings"
+	"romano.com/proto"
+	"romano.com/mongo"
+	"romano.com/model/fileModel"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/reflection"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/bson/primitive"
+)
+
+type Server struct {
+	port string
+}
+
+const (
+	mongoID string = "_id";
+)
+
+var mongoManager mongo.Controller;
+
+func New(port string) Server {
+	var s Server;
+	s.port = port;
+	mongoManager = mongo.NewController("mongodb://mongo:27017");
+  	return s;
+}
+
+func (s *Server) CreateListener() {
+
+	listener, err := net.Listen("tcp", s.port);
+	if err != nil {
+		panic(err)
+	}
+
+	srv := grpc.NewServer();
+	proto.RegisterAddServiceServer(srv, s);
+	reflection.Register(srv);
+
+	fmt.Println("GRpc run on" + s.port);
+	if e := srv.Serve(listener); e != nil {
+		fmt.Println("GRpc not run: ", err.Error());
+	}
+}
+
+func (s *Server) Create(ctx context.Context, req *proto.CreateReq) (*proto.FileRes, error) {
+	owner, name, path, isFolder := req.GetFile().Owner, req.GetFile().Name, req.GetFile().Path, req.GetFile().IsFolder;
+	file := fileModel.New(owner, name, path, isFolder);
+	decoded, err := file.Insert(mongoManager.Client)
+
+	fmt.Println(decoded.ID.Hex());
+
+	if err != nil {
+		return nil, err
+	}
+
+	return &proto.FileRes{
+		File: &proto.File{
+			Id:       decoded.ID.Hex(),
+			Owner:    decoded.Owner,
+			Path:     decoded.Path,
+			Name:     decoded.Name,
+			IsFolder: decoded.IsFolder,
+		},
+	}, nil
+}
+
+func (s *Server) Read(ctx context.Context, req *proto.ReadReq) (*proto.FileRes, error) {
+	owner := req.GetOwner();
+	oid, err := primitive.ObjectIDFromHex(req.GetId())
+	if err != nil {
+		return nil, fmt.Errorf("InvalidArgument", fmt.Sprintf("Could not convert to ObjectId: %v", err))
+	}
+
+	decoded, err := fileModel.FinedOne(mongoManager.Client, bson.M {mongoID : oid} )
+	
+	if strings.Compare(decoded.Owner, owner) == 0 {
+		return nil, fmt.Errorf("Permition", fmt.Sprintf("Permition denaid %s: %v", req.GetOwner(), err));
```

Comment:
Typo, should be `permittion` and `denied`.

Reply:
Also as I commented earlier, this is an incorrect use of `fmt.Errorf` and there are more places, please change them, you can use the command `go vet ./...` in your project in the terminal and it should show you where these problems are.